pkg/operator/ceph/object/controller.go

```diff
@@ -532,35 +537,76 @@ func (r *ReconcileCephObjectStore) getMultisiteResourceNames(cephObjectStore *ce
 
 func (r *ReconcileCephObjectStore) reconcileCOSIUser(cephObjectStore *cephv1.CephObjectStore) (reconcile.Result, error) {
 	// Create COSI user and secret
-	userConfig := generateCOSIUserConfig()
-	var user admin.User
+	cosiUserConfig := generateCOSIUserConfig()
+	var cosiUser admin.User
 
 	// Create COSI user
 	objCtx, err := NewMultisiteContext(r.context, r.clusterInfo, cephObjectStore)
+
+	logger.Debugf("creating cosi user object %q for object store %q", cosiUserConfig.ID, objCtx.Name)
+
 	if err != nil {
 		return reconcile.Result{}, errors.Wrapf(err, "failed to get object context")
 	}
 
-	adminOpsCtx, err := newMultisiteAdminOpsCtxFunc(objCtx, &cephObjectStore.Spec)
-	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to get admin ops API context")
-	}
+	if cephObjectStore.Spec.IsExternal() {
+		adminOpsCtx, err := newMultisiteAdminOpsCtxFunc(objCtx, &cephObjectStore.Spec)
```

Comment:
So we need to use the admin api for external clusters, but radosgw-admin for local creation?

Reply:
I'd prefer not do this and always use the admin ops API, even if it takes longer for us to be able to successfully use the commands. We have so many problems with `radosgw-admin` that I'd like to use it as infrequently as possible.